src/components/MapPage/MapPage.jsx

```diff
@@ -1,45 +1,37 @@
-import { useState, useEffect, useMemo } from 'react';
+import { useMemo } from 'react';
 import Peripleo, { Map, Controls, ZoomControl } from '@peripleo/peripleo';
 import {
   AuthorSelect,
   ITSBStore,
   ITSBTooltip,
   ItinerariesLayer,
   IntersectionsLayer,
-  MapModeSwitch,
   MonthRangeInput,
-} from './components';
+} from '../../components';
+import { useLoaderData, useMatch } from 'react-router-dom';
+import './MapPage.css';
 
 const fetchData = (url) => fetch(url).then((res) => res.json());
 
-export function App() {
-  const [authors, setAuthors] = useState();
+export const mapLoader = async () =>
+  await Promise.all([
+    fetchData('data/authors.json'),
+    fetchData('data/places.json'),
+    fetchData('data/itineraries.json'),
+  ]);
 
-  const [places, setPlaces] = useState();
+export function MapPage() {
+  const [authors, places, itineraries] = useLoaderData();
 
-  const [itineraries, setItineraries] = useState();
+  const loaded = authors && places && itineraries;
 
-  const [mode, setMode] = useState('trajectories');
+  const isTrajectories = useMatch('trajectories');
 
   const layer = useMemo(
-    () => (mode === 'trajectories' ? ItinerariesLayer : IntersectionsLayer),
-    [mode]
+    () => (isTrajectories ? ItinerariesLayer : IntersectionsLayer),
```

Comment:
I think you are right. `useMemo` doesn't add much benefit here. It's just a single assignment of the `layer` const, i.e. not worth the overhead. Also, I don't think the MapPage component would re-render on any other occasion than page navigation. (In which case `isTrajectories` would change, too.) IMO good for removal!